### DISCUSSION TITLE
Attach a file to a new record

### DISCUSSION THREAD

**Discussion by DrMarkX** [Snow-Shell/servicenow-powershell#215] [Q&A] [answered]
I see there is a command Export-ServiceNowAttachment.  Is there a way to attach a file to a newly created record or update a record with an attachment?  If the latter, how would I get the sys_id of the newly created record (other than retrieving via the short description and possibly the date created)?

**Comment by gdbarron** [comment, marked as answer]
New-ServiceNowRecord has a `-PassThru` parameter to return the newly created record.  If you want to create and add a file in one step, you can pipe one to the other, eg. `New-ServiceNowIncident @params -PassThru | Add-ServiceNowAttachment -File file01.txt`

**Reply by DrMarkX** [reply]
So smooth!  Thank you.
